pkg/controller/worker/machines.go

```diff
@@ -270,11 +266,40 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 		// Availability Zones
 		var zoneCount = len(pool.Zones)
 		for zoneIndex, zone := range pool.Zones {
+			var (
+				subnetName     string
+				workerPoolHash string
+			)
+
+			if infrastructureStatus.Networks.Topology == azureapi.TopologyZonal {
+				subnetIndex, nodesSubnet, err := azureapihelper.FindSubnetByPurposeAndZone(infrastructureStatus.Networks.Subnets, azureapi.PurposeNodes, zone)
+				if err != nil {
+					return err
+				}
+				subnetName = nodesSubnet.Name
+				if subnetIndex == 0 {
```

Comment:
This is just required as we keep during network layout migration the machines assigned to existing subnet, right?

Reply:
Yes. So the subnetName was not used previously when calculating the workerpool hash.
This change requires that:
a) machines in zone["0"] (the one keeping the original subnet) are correctly placed and thus don't need any change in their hash (avoid triggering rolling updates).
b) all other machines that reside in zones but may have been using the previously single subnet to be rolled to use the new subnets.

The index here is used to make the distinction on whether to use or not the subnet as part of the hash.